Added player hit-boxes

gmanjames opened the pull request:
Added two FlxObjects to serve as top and bottom hit-boxes for the player. We have tested in class using these to address some difficulties with colliding properly with blocks. Primarily, these will be of use to @Cortillaen

gmanjames committed:
Added player hitboxes. Broken

gmanjames committed:
Added multiple player hitboxes

SamBumgardner commented on source/Player.hx:
```diff
@@ -51,6 +56,13 @@ import flixel.input.keyboard.FlxKey;
+    // Multiple hitbox support
+    hitBoxComponents = new FlxTypedGroup<FlxObject>(2);
+    topBox = new FlxObject(X + 1, Y, width - 2, 3);
+    btmBox = new FlxObject(X + 1, Y + height - 4, width - 2, 3);
```
Let's get these numbers replaced with variables, if possible.

SamBumgardner commented on source/Player.hx:
```diff
@@ -133,4 +146,16 @@ import flixel.input.keyboard.FlxKey;
+
+    topBox.x = btmBox.x = hitBoxX;
+    topBox.y = y + hitBoxOffset;
+    btmBox.y = y + height - hitBoxOffset - hitBoxHeight;
```
By this math, these hitboxes are inside the player's main hitbox. 
I don't think this is the desired behavior, because it could lead to instances where the player overlaps with an enemy below but the bottom hitbox does not.

SamBumgardner commented on source/Player.hx:
```diff
@@ -133,4 +146,16 @@ import flixel.input.keyboard.FlxKey;
     return FlxG.keys.anyPressed([FlxKey.Z]);
   }
+
+  private function updateHitBoxes():Void
```
Don't forget to add documentation to all functions!

SamBumgardner reviewed (changes requested):
Looks pretty straightforward overall. Once my comments are addressed/responded to, we'll probably be good to merge!

gmanjames committed:
Address pull request issues.

Added documentation to Player method. Moved a magic number to a variable for use in Player

SamBumgardner commented:
Don't forget about the hitboxes-inside-the-player thing

gmanjames commented:
They should extend to either edge of the player now that the offset has been removed

SamBumgardner commented:
Okay, it looks good to me. Merging in now!
